src/honey_badger/epoch_state.rs

```diff
@@ -38,8 +38,10 @@ impl<N: NodeIdT> SigningState<N> {
     fn handle_message(&mut self, sender_id: &N, msg: ts::Message) -> ts::Result<ts::Step<N>> {
         match self {
             SigningState::None => {
-                let fault_kind = FaultKind::UnexpectedSignatureShare;
-                Ok(Fault::new(sender_id.clone(), fault_kind).into())
+                let fault_kind = ts::FaultKind::UnexpectedSignatureShare;
+                let mut step = ts::Step::default();
+                step.fault_log.append(sender_id.clone(), fault_kind);
+                Ok(step)
```

Comment:
Why was this change necessary? Shouldn't `step` still implement `From<Fault<…>>`?